### PR TITLE
Add auto construct container fixes

### PR DESCRIPTION
**Pull request by dagnir**
- Check for auto construct containers on paginator termination
   condition
 - Enable autoconstruct list for DDB
 - Fix broken tests

## Description

Add auto construct container fixes

## Motivation and Context
Integration test fixes.

## Testing
unit and integration tests (minus the mapper for integration tests).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: test/dynamodbmapper-v1/src/it/java/software/amazon/awssdk/services/dynamodb/mapper/GenerateCreateTableRequestIntegrationTest.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import java.util.Set;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
```
Comment (varunnvs92) [inline test/dynamodbmapper-v1/src/it/java/software/amazon/awssdk/services/dynamodb/mapper/GenerateCreateTableRequestIntegrationTest.java]: Not used.

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/paginators/PaginatorsClassSpec.java [hunk 2]
```diff
@@ -165,21 +167,24 @@ protected MemberModel memberModelForResponseMember(String input) {
...
-            // If there is no more_results token, then output_token will be a single value
-            body = String.format("return %s.%s != null",
-                                 PREVIOUS_PAGE_METHOD_ARGUMENT,
-                                 fluentGetterMethodsForOutputToken().get(0));
+                                 fluentGetterMethodForResponseMember(paginatorDefinition.getMoreResults()))
+                    .build();
         }
+        // If there is no more_results token, then output_token will be a single value
+        return CodeBlock.builder()
+                .add("return $1N.$2L != null && !$3T.class.isInstance($1N.$2L) && !$4T.class.isInstance($1N.$2L)",
```
Comment (varunnvs92) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/paginators/PaginatorsClassSpec.java]: So our assumption is "Output token member will not be a list or map" when more_results key is missing in paginators file. Is that correct?

Comment (dagnir) [reply codegen/src/main/java/software/amazon/awssdk/codegen/poet/paginators/PaginatorsClassSpec.java]: No, it doesn't make any assumptions about the output token. We treat auto construct lists and maps as null so it's essentially a null check here just like it was original.

**Comment by dagnir** [comment]
Ran integration tests after I rebased against master (previously 56ab06e09267797d1b36eed676a20c84a99c51e5)  and S3 failed with what appears to be unrelated errors. 

S3 tests also failing with same failures when using current HEAD of master.

**Pull request merged**